File: fix_script.py
import os

file_path = r'c:\Users\Admin\Documents\GitHub\tienda\script.js'

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Fix 1: Purchases tab syntax (around line 4554)
# We look for the IIFE closing and fix it.
for i in range(len(lines)):
    if 'REGISTRAR REPOSICIÓN DE STOCK' in lines[i] and 'button' in lines[i-1]:
        # This is the button. The IIFE closing is above it.
        # We need to find the specific pattern.
        pass

# Actually, I'll just look for the specific lines.
for i in range(len(lines)):
    if ');' in lines[i] and '})()' in lines[i+1] and i > 4500 and i < 4600:
        lines[i] = ' ' * 52 + ')\n'
        lines[i+1] = ' ' * 48 + '})()}\n'
        break

# Fix 2: Remove duplicate footer/modals and fix hierarchy
# We'll remove the block between line 6645 and 6822.
# And we'll re-insert the privacy view and a clean footer into a better spot.

# Actually, it's easier to just find the markers.
start_duplicate = -1
end_duplicate = -1
for i in range(len(lines)):
    if 'FOOTER PROFESIONAL' in lines[i]:
        if start_duplicate == -1:
            start_duplicate = i
        else:
            end_duplicate = i # This is the start of the second footer

if start_duplicate != -1 and end_duplicate != -1:
    # Remove everything from the first footer up to the end of the first modal block
    # We'll assume the first block ends at line 6822 (approximately)
    # Let's find the ')}' that closes the manual sale modal before the second footer.
    block_end = -1
    for i in range(end_duplicate, start_duplicate, -1):
        if ')}' in lines[i] and '</main>' in lines[i+1]:
            block_end = i
            break
    
    if block_end != -1:
        # Remove the block
        del lines[start_duplicate:end_duplicate]

# Final Cleanup: Ensure App ends correctly
# (This is more complex, but we'll try to reach a stable state)

with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(lines)

print("File updated successfully.")

File: fix_script.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('c:', 'Users', 'Admin', 'Documents', 'GitHub', 'tienda', 'script.js');
let content = fs.readFileSync(filePath, 'utf8');

// Fix 1: Purchases tab syntax (around line 4554)
// We look for the IIFE return and closing and fix it.
// Original:
// </div>
// );
// })()}
// We match with a regex that is flexible with indentation.
content = content.replace(/(<\/div>\s+)\);\s+(\s+\}\)\(\))/g, '$1)\n$2');

// Fix 2: Remove First Footer/Modal Block and Second main Tag
// We want to find the section from the first footer up to the second main tag.
// First footer starts around 6645.
// Second footer starts around 6826.

const firstFooterIndex = content.indexOf('{view !== \'admin\' && view !== \'login\' && view !== \'register\' && (');
const secondFooterIndex = content.lastIndexOf('{view !== \'admin\' && view !== \'login\' && view !== \'register\' && (');

if (firstFooterIndex !== secondFooterIndex && firstFooterIndex !== -1) {
    // We found duplicates. 
    // We'll also find the </main> tag that's between them.
    const redundantCloseMain = content.indexOf('</main>', firstFooterIndex);

    if (redundantCloseMain !== -1 && redundantCloseMain < secondFooterIndex) {
        // We'll remove from firstFooterIndex to the point just before the second footer.
        // BUT we need to make sure we don't break the admin view closing.

        // Actually, let's just replace the WHOLE problematic section with something clean.
        // We'll find the AccessDenied closing (around 6641).
        const accessDeniedEnd = content.indexOf('<AccessDenied onBack={() => setView(\'store\')} />\r\n                    ))}', 4103);
        // Wait! Indentation might vary. 
        // Let's use a regex for the AccessDenied closing.
        const accessDeniedPattern = /<AccessDenied onBack=\{\(\) => setView\('store'\)\} \/>\s+\)\s+\)\s+\}/;
        const match = content.match(accessDeniedPattern);

        if (match) {
            const startOfEnd = match.index + match[0].length;
            // From here up to the start of the SECOND footer, everything is mess.
            content = content.slice(0, startOfEnd) + '\n                )}\n            </main>\n\n            ' + content.slice(secondFooterIndex);
        }
    }
}

// Fix 3: Restore Privacy Policy view (if missing)
if (content.indexOf('view === \'privacy\'') === -1) {
    const mainEndIndex = content.lastIndexOf('</main>');
    if (mainEndIndex !== -1) {
        const privacyView = `
            {/* 8. VISTA POLÍTICA DE PRIVACIDAD */}
            {view === 'privacy' && (
                <div className="max-w-4xl mx-auto py-20 px-6 animate-fade-up">
                    <div className="glass p-12 rounded-[3rem] border border-slate-800">
                        <div className="prose prose-invert max-w-none">
                            <h1 className="text-5xl font-black mb-12 tracking-tighter italic">
                                Política de <span className="text-cyan-500 text-6xl">Privacidad</span>
                            </h1>
                            <p className="text-slate-400 text-lg leading-relaxed">
                                En <strong>{settings?.storeName || 'SUSTORE'}</strong>, valoramos tu privacidad y nos comprometemos a proteger tus datos personales. Esta política describe cómo recolectamos, usamos y resguardamos tu información.
                            </p>
                            <h2 className="text-2xl font-bold text-white mt-12 mb-6">1. Información Recolectada</h2>
                            <p className="text-slate-500 leadind-relaxed">
                                Recolectamos datos básicos como nombre, correo electrónico y número de teléfono únicamente cuando te registras o realizas un pedido para procesar tu compra correctamente.
                            </p>
                            <h2 className="text-2xl font-bold text-white mt-12 mb-6">2. Uso de los Datos</h2>
                            <p className="text-slate-500 leadind-relaxed">
                                Tu información se utiliza exclusivamente para:
                            </p>
                            <ul className="list-disc pl-6 text-slate-500 space-y-2">
                                <li>Gestionar tus pedidos y entregas.</li>
                                <li>Enviar actualizaciones sobre el estado de tu compra.</li>
                                <li>Mejorar nuestros servicios y experiencia de usuario.</li>
                            </ul>
                            <h2 className="text-2xl font-bold text-white mt-12 mb-6">3. Seguridad</h2>
                            <p className="text-slate-500 leadind-relaxed">
                                Implementamos medidas de seguridad robustas y encriptación de datos para asegurar que tu información esté protegida contra accesos no autorizados.
                            </p>
                            <h2 className="text-2xl font-bold text-white mt-12 mb-6">4. Contacto</h2>
                            <p className="text-slate-500 leadind-relaxed mb-12">
                                Si tienes dudas sobre nuestra política de privacidad, contáctanos a <span className="text-cyan-400">{settings?.contactEmail || 'soporte@sustore.sf'}</span>.
                            </p>
                            <button onClick={() => setView('store')} className="px-10 py-4 bg-slate-900 hover:bg-slate-800 text-white rounded-2xl font-bold transition flex items-center gap-3 border border-slate-700">
                                <ArrowLeft className="w-5 h-5" /> Volver a la Tienda
                            </button>
                        </div>
                    </div>
                </div>
            )}
        `;
        content = content.slice(0, mainEndIndex) + privacyView + content.slice(mainEndIndex);
    }
}

fs.writeFileSync(filePath, content);
console.log('File script.js optimized and fixed.');
